io-engine/src/rebuild/rebuild_job.rs

```diff
@@ -395,12 +398,20 @@ impl<'n> RebuildJob<'n> {
             &mut copy_buffer
         };
 
-        source_hdl
-            .read_at(blk * self.block_size, copy_buffer)
-            .await
-            .context(ReadIoFailed {
-                bdev: &self.src_uri,
-            })?;
+        source_hdl.set_read_mode(ReadMode::UnwrittenFail);
+
+        let res = source_hdl.read_at(blk * self.block_size, copy_buffer).await;
+
+        if let Err(CoreError::ReadingUnallocatedBlock {
```

Comment:
It will still fail. Probably, it would be a good idea to limit read range to be less or equal to cluster size when SPDK_NVME_IO_FLAGS_UNWRITTEN_READ_FAIL is specified.

Reply:
But even with the same size it'd be tricky because if it's not aligned we're back to error?